File: MPI/MPI_day1_2022/pi_serial.c
#include <stdio.h>
#include <math.h>

int main(int argc, char *argv[])
{
  int n, i;
  double PI25DT = 3.141592653589793238462643;
  double pi, h, sum, x;
  while (1) {
     printf("Enter the number of intervals: (0 quits) ");
     scanf("%d",&n);
     if (n == 0)
         break;
     else {
         h   = 1.0 / (double) n;
         sum = 0.0;
         for (i = 1; i <= n; i++) {
             x = h * ((double)i - 0.5);
             sum += (4.0 / (1.0 + x*x));
         }
         pi = h * sum;
         printf("pi is approximately %.16f, Error is %.16f\n",
                    pi, fabs(pi - PI25DT));
     }
  }
  return 0;
}

File: MPI/MPI_day1_2022/pi_serial.f90
     program main
     double precision  PI25DT
     parameter        (PI25DT = 3.141592653589793238462643d0)
     double precision  pi, h, sum, x, f, a
     integer n, i
!                                function to integrate
     f(a) = 4.d0 / (1.d0 + a*a)

     do
        print *, 'Enter the number of intervals: (0 quits) '
        read(*,*) n
!                                check for quit signal
        if (n .le. 0) exit
!                                calculate the interval size
        h = 1.0d0/n
        sum  = 0.0d0
        do i = 1, n
           x = h * (dble(i) - 0.5d0)
           sum = sum + f(x)
        enddo
!
        pi = h * sum
        print *, 'pi is ', pi, ' Error is', abs(pi - PI25DT)
!
     enddo
!
     end
